core/src/main/scala/org/apache/spark/storage/BlockManagerId.scala

```diff
@@ -132,10 +133,15 @@ private[spark] object BlockManagerId {
     getCachedBlockManagerId(obj)
   }
 
-  val blockManagerIdCache = new ConcurrentHashMap[BlockManagerId, BlockManagerId]()
+  val blockManagerIdCache = CacheBuilder.newBuilder()
+    .maximumSize(500)
```

Comment:
here i set 500
since `blockmanagerId` about `48B` per object.
I do not use spark conf since it is not convenient to get spark conf for historyserver when use BlockManagerId

Reply:
I'm ok to hardcode it for now.

Reply:
I‘m +1 on hardcode this, but we shall also explain in the comment the reason why this number is chosen.

Reply:
Actually i think 500 executors can handle most applications.And for historyserver it is no need to cache too much `BlockManagerId`.If we set this number as 50 the max size of cache will below 30KB.
Agree with that? @jiangxb1987 If ok i will update documentation.

Reply:
Please feel free to update any comment, and I would set the default value to 10000 since I think cost of 500KB memory is feasible.

Reply:
Thanks @jiangxb1987 i have updated the comment